file_organizer_v2/src/file_organizer/services/deduplication/extractor.py

```
@@ -0,0 +1,339 @@
+        Args:
+            file_path: Path to document file
+
+        Returns:
+            Extracted text content
+
+        Raises:
+            ValueError: If file format not supported
+            IOError: If file cannot be read
+        """
+        if not file_path.exists():
+            raise IOError(f"File not found: {file_path}")
```

Ruff (0.14.13)

Replace aliased errors with `OSError`

Replace `IOError` with builtin `OSError`

(UP024)